### PR TITLE
Created ProductItem Page

### PR DESCRIPTION
**Pull request by msaivasanth**
Added a product item page which shows the details of the single product in a separate page

### CODE REVIEW

File: src/Components/ProductItem.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        images: string[],
+        thumbnail: string
+    }
+    const [productDetails, setProductDetails] = useState<productDetailsProps | null>(null);
+    useEffect(() => {
+        getDetails()
+    }, [])
+
+    const getDetails = async () => {
+        const response = await fetch(`https://dummyjson.com/products/${id}`)
```
Comment (syedzeeshan1) [inline src/Components/ProductItem.tsx]: move to a service file

Comment (syedzeeshan1) [reply src/Components/ProductItem.tsx]: also, shouldn't each api call have the token in the header? if there is no valid token, api data should not be retrieved

**Review by syedzeeshan1 — CHANGES_REQUESTED** [review]

File: src/Components/ProductItem.tsx (hunk 1 above)
Comment (msaivasanth) [reply src/Components/ProductItem.tsx]: Sir, the api is giving the information with or without the token as a header, I think, if we check the token that exists in local storage or not before doing the api call. Then the we will call the api if token exists or else we will throw an alert

Comment (syedzeeshan1) [reply src/Components/ProductItem.tsx]: maybe use this? https://dummyjson.com/docs/auth#me
so basically check if token from local storage is valid then do the call

Comment (msaivasanth) [reply src/Components/ProductItem.tsx]: Okay sir, now I will add this change in product item page and can I do this change in  home page also, before fetching all the product details.

File: src/ProductsContext/ProductState.tsx [hunk 2]
```diff
@@ -28,7 +28,27 @@ const ProductState: React.FC<ProductStateProps> = (props: any) => {
     const json = await response.json();
     return json
   }
-  const value: any = { getProducts, products, setName, setPassword, name, password, handleLogin };
+
+  const handleGetDetails = async (id: number) => {
+    const check = await fetch('https://dummyjson.com/auth/me', {
```
Comment (syedzeeshan1) [inline src/ProductsContext/ProductState.tsx]: you will be using this logic with every api. so why not move this auth/me into a separate fn?
that way, for every api call you do in service files, you presumably start with this step.

make sure all apis after login are using this!

**Review by syedzeeshan1 — APPROVED** [review]
Overall great work! Small change you can address and merge! good job

**Comment by msaivasanth** [comment]
Made the mentioned changes. Please check and give me the final confirmation for merging with main branch

**Review by syedzeeshan1 — APPROVED** [review]
🚢

**Pull request merged**